src/QueryBuilder/ResultViewer/DynamicTable/DynamicTable.js

```diff
@@ -14,6 +14,10 @@ function getCellValue(row, property) {
       : <FormattedMessage id="ui-plugin-query-builder.options.false" />;
   }
 
+  if (property.dataType.dataType === DATA_TYPES.DateType) {
+    return row[property.property] ? <FormattedDate value={row[property.property]} /> : '';
+  }
+
```

Comment:
we didn't do this for nested fields yet (although I don't think we have any applicable)